ui/src/components/DataTable/index.jsx

```diff
@@ -40,22 +41,33 @@ export default class DataTable extends Component {
     renderRow: func.isRequired,
     /**
      * A function to execute when a column header is clicked.
-     * Will receive a single argument which is the column name.
+     * Will receive a single argument
+     * which is the header object for the corresponding column.
      * This can be used to handle sorting.
      */
     onHeaderClick: func,
     /**
      * A header name to sort on.
      */
-    sortByHeader: string,
+    sortByLabel: string,
     /**
      * The sorting direction.
      */
     sortDirection: oneOf(['desc', 'asc']),
     /**
-     * A list of header names to use on the table starting from the left.
+     * An list of objects. Each element contains a label
+     * representing a header name to use on the table starting from the left,
+     * an id which is a camel-cased version of the header name used in
+     * the sorting function to sort column contents
+     * and a type, a string identifying the data type of the column contents.
      */
-    headers: arrayOf(string),
+    headers: arrayOf(
+      shape({
+        id: string,
```

Comment:
`A identifier for the header. This is used to identify which column was clicked for example.`